lib/kafka_ex/protocol/fetch.ex

```diff
@@ -73,6 +73,13 @@ defmodule KafkaEx.Protocol.Fetch do
   defp parse_message_set([last|_] = list, _) do
     {:ok, Enum.reverse(list), last.offset}
   end
+  defp parse_message_set(_, << offset :: 64, msg_size :: 32, rest :: binary >>) do
```

Comment:
Rather than have a fall-through try to replicate the original clause error, how about having a clause specifically for what we're trying to catch?

```
  defp parse_message_set(_, << offset :: 64, msg_size :: 32, rest :: binary >>) when byte_size(rest) < msg_size do
    raise RuntimeError, "Too little data fetched at offset #{offset}. Message size #{msg_size} but only got another #{byte_size rest} bytes! Try increasing max_bytes to at least #{msg_size + 12}."
  end
```

Also can I request we clean up the error message a little?  How about `"Insufficient data fetched at offset #{offset}.  Message size is #{msg_size} but we only recieved #{byte_syze(rest)} bytes.  Increase the value of max_bytes." (Unless you had a specific reason to suggest increasing it by 12 bytes?)

Reply:
The 12 byte increase takes the `offset` and `msg_size` into account. When I tried consuming with a `max_bytes` of 60000, there were 59988 bytes of message received. Adding 12 to the `msg_size` should allow the full message plus offset/size header to be received. Might be too specific, as it's unlikely that that particular size is going to prevent future errors (it really depends on the broker's max setting, as that's the size that will be rejected on publishing).